Develop upstream sync 250714

mmakevic-amd opened the pull request:
List of disabled tests:
```
//xla/backends/gpu/codegen/triton:fusion_emitter_device_legacy_port_test
         TritonGemmTest.SplitLHSOutputTransposeAloneIsNotFused
         TritonEmitterTest.RocmWarpSizeIsSetCorrectly

//xla/backends/gpu/codegen/triton:fusion_emitter_int4_device_test
          TritonTest.FuseSubchannelDequantizationWithTranspose
          TritonTest.FuseSubchannelDequantization
          TritonTest.FuseChannelDequantization
          TritonTest.FuseMultiplyInPrologue
          TritonTest.NonstandardLayoutWithManyNonContractingDimsReversedLayout

//xla/backends/gpu/codegen/triton:fusion_emitter_legacy_int4_device_test
          TritonTest.FuseSubchannelDequantizationWithTranspose
          TritonTest.FuseSubchannelDequantization
          TritonTest.FuseChannelDequantization
          TritonTest.FuseBroadcastBitcastMultiplyInPrologue
          TritonTest.FuseMultiplyInPrologue
          TritonTest.NonstandardLayoutWithManyNonContractingDimsReversedLayout

//xla/backends/gpu/codegen/triton:fusion_emitter_parametrized_test
          TritonNormalizationTest.CanFuseAndEmitDiamondWithBF16Converts

//xla/tests:multioutput_fusion_test
          MultiOutputFusionTest.MultiOutputReduceFusionMajorWithExtraOutput

//xla/tools/hlo_opt:tests/gpu_hlo_llvm.hlo

//xla/service/gpu/transforms:triton_fusion_numerics_verifier_test
         TritonFusionNumericsVerifierTest.VerifyNestedGemmNumerics
         TritonFusionNumericsVerifierTest.CompilationSucceedsEvenIfKernelWillSpillRegisters
         TritonFusionNumericsVerifierTest.VerifyThatDisablingTritonIsFast

//xla/backends/gpu/codegen/emitters/tests:reduce_row/mof_scalar_variadic.hlo.test

//xla/backends/gpu/codegen/emitters/tests:reduce_row/side_output_broadcast.hlo.test
```

tensorflower-gardener committed:
#HLODiff Add bipartite matching to GreedyTopDownMatcher.

PiperOrigin-RevId: 780307215

tensorflower-gardener committed:
Update XNNPACK to get new APIs

PiperOrigin-RevId: 780310627

tomnatan30 committed:
[StableHLO] Bump the 4 week compatibility version to 1.11.0

PiperOrigin-RevId: 780330580

tensorflower-gardener committed:
Don't try to handle batched dot products.

XNNPACK's `xnn_define_batch_matrix_multiply` can't handle this case, because it interprets the batch dimensions as matrix dimensions.

Previously, we accidentally avoided this bug due to the heuristic to avoid using XNNPACK when we want to tile K, but we should handle this case explicitly.

PiperOrigin-RevId: 780330672

emilyfertig committed:
[JAX] Use experimental DCN transfer library as a fallback for PjRt-IFRT cross-host device transfers when the PjRt plugin doesn't implement the cross-host transfer APIs.

PiperOrigin-RevId: 780333956

hhb committed:
Call get_logical_on_device_shape within thread pool.

PiperOrigin-RevId: 780357026

tensorflower-gardener committed:
Use the output shape from executables when minting error buffers

PiperOrigin-RevId: 780373769

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780396029

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780413917

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780415084

tensorflower-gardener committed:
Set XNN_FLAG_SLINKY_STATIC_BOUNDS in xnn_fusion_thunk.

PiperOrigin-RevId: 780419314

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780421697

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780452074

tensorflower-gardener committed:
Update GraphDef version to 2282.

PiperOrigin-RevId: 780467978

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2025-07-08

PiperOrigin-RevId: 780468040

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780468639

dependabot[bot] committed:
PR #28591: Bump github/codeql-action from 3.29.1 to 3.29.2

Imported from GitHub PR https://github.com/openxla/xla/pull/28591

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.1 to 3.29.2.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a href="https://github.com/github/codeql-action/releases">github/codeql-action's releases</a>.</em></p>
<blockquote>
<h2>v3.29.2</h2>
<h1>CodeQL Action Changelog</h1>
<p>See the <a href="https://github.com/github/codeql-action/releases">releases page</a> for the relevant changes to the CodeQL CLI and language packs.</p>
<h2>3.29.2 - 30 Jun 2025</h2>
<ul>
<li>Experimental: When the <code>quality-queries</code> input for the <code>init</code> action is provided with an argument, separate <code>.quality.sarif</code> files are produced and uploaded for each language with the results of the specified queries. Do not use this in production as it is part of an internal experiment and subject to change at any time. <a href="https://redirect.github.com/github/codeql-action/pull/2935">#2935</a></li>
</ul>
<p>See the full <a href="https://github.com/github/codeql-action/blob/v3.29.2/CHANGELOG.md">CHANGELOG.md</a> for more information.</p>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a href="https://github.com/github/codeql-action/blob/main/CHANGELOG.md">github/codeql-action's changelog</a>.</em></p>
<blockquote>
<h1>CodeQL Action Changelog</h1>
<p>See the <a href="https://github.com/github/codeql-action/releases">releases page</a> for the relevant changes to the CodeQL CLI and language packs.</p>
<h2>[UNRELEASED]</h2>
<p>No user facing changes.</p>
<h2>3.29.2 - 30 Jun 2025</h2>
<ul>
<li>Experimental: When the <code>quality-queries</code> input for the <code>init</code> action is provided with an argument, separate <code>.quality.sarif</code> files are produced and uploaded for each language with the results of the specified queries. Do not use this in production as it is part of an internal experiment and subject to change at any time. <a href="https://redirect.github.com/github/codeql-action/pull/2935">#2935</a></li>
</ul>
<h2>3.29.1 - 27 Jun 2025</h2>
<ul>
<li>Fix bug in PR analysis where user-provided <code>include</code> query filter fails to exclude non-included queries. <a href="https://redirect.github.com/github/codeql-action/pull/2938">#2938</a></li>
<li>Update default CodeQL bundle version to 2.22.1. <a href="https://redirect.github.com/github/codeql-action/pull/2950">#2950</a></li>
</ul>
<h2>3.29.0 - 11 Jun 2025</h2>
<ul>
<li>Update default CodeQL bundle version to 2.22.0. <a href="https://redirect.github.com/github/codeql-action/pull/2925">#2925</a></li>
<li>Bump minimum CodeQL bundle version to 2.16.6. <a href="https://redirect.github.com/github/codeql-action/pull/2912">#2912</a></li>
</ul>
<h2>3.28.19 - 03 Jun 2025</h2>
<ul>
<li>The CodeQL Action no longer includes its own copy of the extractor for the <code>actions</code> language, which is currently in public preview.
The <code>actions</code> extractor has been included in the CodeQL CLI since v2.20.6. If your workflow has enabled the <code>actions</code> language <em>and</em> you have pinned
your <code>tools:</code> property to a specific version of the CodeQL CLI earlier than v2.20.6, you will need to update to at least CodeQL v2.20.6 or disable
<code>actions</code> analysis.</li>
<li>Update default CodeQL bundle version to 2.21.4. <a href="https://redirect.github.com/github/codeql-action/pull/2910">#2910</a></li>
</ul>
<h2>3.28.18 - 16 May 2025</h2>
<ul>
<li>Update default CodeQL bundle version to 2.21.3. <a href="https://redirect.github.com/github/codeql-action/pull/2893">#2893</a></li>
<li>Skip validating SARIF produced by CodeQL for improved performance. <a href="https://redirect.github.com/github/codeql-action/pull/2894">#2894</a></li>
<li>The number of threads and amount of RAM used by CodeQL can now be set via the <code>CODEQL_THREADS</code> and <code>CODEQL_RAM</code> runner environment variables. If set, these environment variables override the <code>threads</code> and <code>ram</code> inputs respectively. <a href="https://redirect.github.com/github/codeql-action/pull/2891">#2891</a></li>
</ul>
<h2>3.28.17 - 02 May 2025</h2>
<ul>
<li>Update default CodeQL bundle version to 2.21.2. <a href="https://redirect.github.com/github/codeql-action/pull/2872">#2872</a></li>
</ul>
<h2>3.28.16 - 23 Apr 2025</h2>
<ul>
<li>Update default CodeQL bundle version to 2.21.1. <a href="https://redirect.github.com/github/codeql-action/pull/2863">#2863</a></li>
</ul>
<h2>3.28.15 - 07 Apr 2025</h2>
<ul>
<li>Fix bug where the action would fail if it tried to produce a debug artifact with more than 65535 files. <a href="https://redirect.github.com/github/codeql-action/pull/2842">#2842</a></li>
</ul>
<h2>3.28.14 - 07 Apr 2025</h2>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a href="https://github.com/github/codeql-action/commit/181d5eefc20863364f96762470ba6f862bdef56b"><code>181d5ee</code></a> Merge pull request <a href="https://redirect.github.com/github/codeql-action/issues/2957">#2957</a> from github/update-v3.29.2-4c57370d0</li>
<li><a href="https://github.com/github/codeql-action/commit/c77386a9db782647c8e2575da69a3c950786eaca"><code>c77386a</code></a> Fix changelog PR number</li>
<li><a href="https://github.com/github/codeql-action/commit/8d43d4ecec27cc4205b0eaaf2e9b4bf9ee9a305b"><code>8d43d4e</code></a> Update changelog for v3.29.2</li>
<li><a href="https://github.com/github/codeql-action/commit/4c57370d0304fbff638216539f81d9163f77712a"><code>4c57370</code></a> Merge pull request <a href="https://redirect.github.com/github/codeql-action/issues/2935">#2935</a> from github/mbg/interpret-cq-results</li>
<li><a href="https://github.com/github/codeql-action/commit/2830b750e5012e0a57cb63888cd5720f2326ca5c"><code>2830b75</code></a> Add changelog entry</li>
<li><a href="https://github.com/github/codeql-action/commit/aa72ddaeada556e7d763c9a0afb01f2c2a365e1c"><code>aa72dda</code></a> Merge branch 'main' into mbg/interpret-cq-results</li>
<li><a href="https://github.com/github/codeql-action/commit/65d1e45f0ba420207efc0f1f6d90c63dcbc97551"><code>65d1e45</code></a> Rename <code>SARIF_UPLOAD_ENDPOINT</code> members</li>
<li><a href="https://github.com/github/codeql-action/commit/362ebf85dad6ee3df420db2cec285490b289a61f"><code>362ebf8</code></a> Check both SARIF files in <code>quality-queries.yml</code> test</li>
<li><a href="https://github.com/github/codeql-action/commit/10a3e4b17dd8a1cee767213c309bd4b1e8251eab"><code>10a3e4b</code></a> Fix formatting</li>
<li><a href="https://github.com/github/codeql-action/commit/8593ea65e2bf97ec2caa80fb0e464ed8c42c0fae"><code>8593ea6</code></a> Merge pull request <a href="https://redirect.github.com/github/codeql-action/issues/2954">#2954</a> from github/mergeback/v3.29.1-to-main-39edc492</li>
<li>Additional commits viewable in <a href="https://github.com/github/codeql-action/compare/39edc492dbe16b1465b0cafca41432d857bdb31a...181d5eefc20863364f96762470ba6f862bdef56b">compare view</a></li>
</ul>
</details>
<br />

[![Dependabot compatibility score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=github/codeql-action&package-manager=github_actions&previous-version=3.29.1&new-version=3.29.2)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting `@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop Dependabot creating any more for this major version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop Dependabot creating any more for this minor version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)

</details>
Copybara import of the project:

--
908f5234414c30c561fcff2ed8bff5d86c4e3440 by dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>:

Bump github/codeql-action from 3.29.1 to 3.29.2

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.1 to 3.29.2.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/39edc492dbe16b1465b0cafca41432d857bdb31a...181d5eefc20863364f96762470ba6f862bdef56b)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Merging this change closes #28591

PiperOrigin-RevId: 780481889

beckerhe committed:
Add missing roctracer dependency to roctracer_wrapper target.

Otherwise all binaries that depend on roctracer_wrapper will fail to link with some undefined reference errors.

PiperOrigin-RevId: 780482037

vwbaker committed:
Fix ragged contracting mode to handle batch dimensions correctly

Before we were not handling the case correctly and it would crash due to incorrect shape at the end. Now, we properly transpose the group dim to the front of the final result.

Also, add tests to clarify this behavior

PiperOrigin-RevId: 780490452

loislo committed:
[XLA:GPU] keep configs even if they occupy less than the available cores

We should not eliminate configs that under-occupy the available cores.
The strategy leads to the configs with high split_k values and does not take into account cost of the follow up reduction fusion. Also for some kernels we don't accept the configs with some split_k because of the restrictions enforced in the split_k rewriter.

PiperOrigin-RevId: 780496018

akuegel committed:
Remove redundant check in CopyRemover (NFC).

We are comparing shapes for equality with the default equality operator. That
includes comparing layouts, which in turn makes sure that memory spaces are the
same. So it is not necessary to check separately whether we have a device to
host or host to device copy, those would have different memory spaces.

PiperOrigin-RevId: 780498246

basioli-k committed:
[XLA][host offloading] Open source HostOffloadingBuffer.

PiperOrigin-RevId: 780506047

akuegel committed:
[HLO_OPT]: Move CopyInsertion pass registration to hardware dependent passes.

This is needed because it requires AliasInfo, which is backend specific. Also
let GpuAliasInfo take a se::DeviceDescription instead of a pointer, as the life
time of the DeviceDescription in the opt tool is not sufficient, it would be
deleted before the pass is run.

PiperOrigin-RevId: 780515794

olegshyshkov committed:
[XLA:GPU] Reset heroes to roots if we device to use loop emitter.

Even if there are non-trivial roots in the fusion, there are cases we want to fall back to the loop emitter. Loop emitter expects that heroes are always roots of the fusion.

PiperOrigin-RevId: 780527758

allanrenucci committed:
Remove obsolete rewrite from `TFLogEntry::text_message_with_prefix` to `TFLogEntry::ToString`

PiperOrigin-RevId: 780528768

basioli-k committed:
[XLA][host offloading] Open source host offloading allocator interface.

PiperOrigin-RevId: 780535586

tensorflower-gardener committed:
Make RequestGpuInfo work for OpenGL < 3.1 as well

Some enums like the compute shader enums fail to query for OpenGL < 3.1.

PiperOrigin-RevId: 780540412

olegshyshkov committed:
[XLA:GPU] Simplify interfaces in CoalescingAnalysis.

A few NFC simplifications:
* We don't need to pass `HloFusionAdaptor`. It's already available in `HloFusionAnalysis`.
* `ComputeCoalescingForAllOperands` can be a stand-alone function outside of `CoalescingAnalysis`.
* No need to pass `KernelFusionInterface`. It's fairly cheap to reconstruct from the analysis

PiperOrigin-RevId: 780542787

loislo committed:
[XLA:GPU] Do not drop low occupancy configs if the search space is small

Low occupancy configs happen when the output size is small. If the number of configs is small it still makes sense to keep these configs. The win in the compile time will be negligible.

PiperOrigin-RevId: 780577138

WillFroom committed:
[XLA:CPU] Add tile size to WorkDimensions.

PiperOrigin-RevId: 780589961

WillFroom committed:
[XLA:CPU][XLA:GPU] Use nsw rather than nuw

PiperOrigin-RevId: 780603812

tensorflower-gardener committed:
#sdy Move `AddAxisOrMergeInserter` to Shardy OSS utils.

PiperOrigin-RevId: 780611537

jparkerh committed:
Add no-arguments compiler client

Add a version of the GetCApiCompiler that takes no arguments - this checks to make sure there's exactly one platform registered in pjrt, and then uses that platform to generate the compiler.

PiperOrigin-RevId: 780614807

basioli-k committed:
[XLA][host offloading] Open source host offloading layout analysis.

PiperOrigin-RevId: 780635346

sdasgup3 committed:
[Phase Compilation] Part-3: Augment extension with buffer deletion callback.

PiperOrigin-RevId: 780642468

emilyfertig committed:
Reverts e2e1ae8fd027c9bdd6e75181e0d5aaaa6a7a12c6

PiperOrigin-RevId: 780645251

ddunl committed:
Use `std::min` instead of `MIN` in `bipartite_matching.cc`

`MIN` causes problems when used with gcc.

PiperOrigin-RevId: 780646584

tensorflower-gardener committed:
Implement `xnn_scheduler` using `Eigen::ThreadPoolInterface` to redirect XNNPACK parallelism to the XLA thread pool.

Other changes in this CL:
- `DebugOptions::XNN_GRAPH_FUSION_MODE_GREEDY_SLINKY` now indicates that slinky should be used (and controls which kind of thread pool gets created).

PiperOrigin-RevId: 780647272

ddunl committed:
Move `tensorflow/third_party/shardy` to `xla/third_party/shardy`

PiperOrigin-RevId: 780652350

dubstack committed:
[HLO Diff] Precompute HLO values dependencies for each instruction in graph.

The instruction dependencies can be useful not just for bottom-up matcher but other matchers and similarity measures as well.

PiperOrigin-RevId: 780668827

allanrenucci committed:
Upgrade Abseil to [LTS 20250127.1](https://github.com/abseil/abseil-cpp/releases/tag/20250127.1).

Not upgrading to [20250512.1](https://github.com/abseil/abseil-cpp/releases/tag/20250512.1) since it breaks Protobuf. These two libraries probably need to be upgraded together.

PiperOrigin-RevId: 780677104

pizzud committed:
strict_cc_test: Only set --gtest_fail_if_no_test_selected on CI.

Local development is reasonably likely to want to use --gtest_filter to run
only a subset of test cases, and in the case of sharded tests this flag
results in spurious failures.

PiperOrigin-RevId: 780689663

pineapplejuice233 committed:
add supports_cross_host_transfers attribute to the new pjrt gpu client.

PiperOrigin-RevId: 780704596

pifon2a committed:
[XLA:GPU] Add TilingSpace.

PiperOrigin-RevId: 780707413

oToToT committed:
Return kTfLiteDelegateError if delegate_kernel is null

If the delegate kernel failed to Init, we will return a nullptr. However, we didn't check when doing Prepare, which will cause a null pointer dereference crash.

This CL avoids this by properly check and return error before accessing a potentially null pointer.

PiperOrigin-RevId: 780710630

dubstack committed:
Simplify HloComputationGraphMatcher by re-using leaf matcher for constant and params

PiperOrigin-RevId: 780718771

ddunl committed:
Remove all uses of `xla_test_library`

Followup will delete definitions, delete `xla_test_libraries` argument used by other macros.

PiperOrigin-RevId: 780719981

changm committed:
Unnest GetShardingFromNodeDef for readability

PiperOrigin-RevId: 780725631

talumbau committed:
Add (filename, offset, length) for disabled MMAPAllocation ctors

PiperOrigin-RevId: 780727101

tensorflower-gardener committed:
Make stateless methods static in XnnGraphFusion. NFC.

PiperOrigin-RevId: 780741805

SandSnip3r committed:
Support Conditional in HostOffloader.

PiperOrigin-RevId: 780753038

toli-y committed:
Eliminate circular dependency between device assignment and Computation Placement. Use Device Assignment for looking up logical and global ids.

PiperOrigin-RevId: 780757147

tensorflower-gardener committed:
[TFLite] Fix incorrect `FuseBinaryOpWithTransposeConvNoneBias` optimization

This commit fixes folding of `F16` `Constant` from a `BinaryOp` (Add or Sub) into the bias of `TransposeConv`, i.e., from `TransposeConv(NoneBias)->Add/Sub(Constant)` to `TransposeConv(Bias)`, where the `Constant` is converted to the `Bias` in `TransposeConv`.
During conversion, `GetBiasMultiplier` function returns a `Multiplier` of `1.0` or `-1.0` depending on `BinaryOp` type (Add or Sub), i.e., `Bias = Constant * Multiplier`. Therefore, it is required that the `Multiplier` has the same datatype as the `Constant`.

Before the fix, this multiplier's datatype is `F32` regardless of the datatype of the `Constant`. This causes error when the `Constant` is `F16`, and the `MulOp` which multiplies the two values returns all-zero values, resulting in an all-zero `Bias`.

After the fix, the `Multiplier` datatype is determined by the `Constant` datatype and the converted `Bias` values are correct for `F16` case.

PiperOrigin-RevId: 780764250

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780773917

ddunl committed:
Delete `xla_test_library` and associated arguments used by other macros

PiperOrigin-RevId: 780775467

tensorflower-gardener committed:
#HLODiff Add lcs parent similarity to MatchInstructionsWithMultipleCandidates

PiperOrigin-RevId: 780778389

subhankarshah committed:
[XLA:MSA] Use max_size_in_bytes instead of available_heap_size() for memory space coloring. It is incorrect to use available_heap_size() because it means remaining available heap size and want to use total heap size available for MSA which is given by max_size_in_bytes.

PiperOrigin-RevId: 780779230

dubstack committed:
Refactor BiDirectionalMap to fully hide internals and expose a symmetric API.

PiperOrigin-RevId: 780793848

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780840832

dubstack committed:
[HLO Diff] Cache right node bfs traversals to speed up DiceSimiliarity calculations.

PiperOrigin-RevId: 780843435

vsytch committed:
[XLA] Make dumping blazingly fast

Scanning the file system during each dump to check how many module we've dumped can slow things down drastically. Reuse the timestamp map to keep track of this.

PiperOrigin-RevId: 780864260

tensorflower-gardener committed:
Merge pull request #96407 from tensorflow:dependabot/pip/ci/official/requirements_updater/numpy1_requirements/requests-2.32.4

PiperOrigin-RevId: 780873054

tensorflower-gardener committed:
Merge pull request #96408 from tensorflow:dependabot/pip/ci/official/requirements_updater/numpy1_requirements/setuptools-78.1.1

PiperOrigin-RevId: 780886988

akuegel committed:
Migrate from CanShareBuffer hook to AliasInfo.

PiperOrigin-RevId: 780895093

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780898250

hhb committed:
Set c api versions in tfrt_gpu_client.cc

PiperOrigin-RevId: 780898958

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780909178

tensorflower-gardener committed:
Update use of deprecated XNNPACK APIs

PiperOrigin-RevId: 780928593

akuegel committed:
Migrate remaining users of CanShareBuffer hook to AliasInfo (NFC)

Delete CanShareBuffer related code.

PiperOrigin-RevId: 780931035

tensorflower-gardener committed:
Update GraphDef version to 2283.

PiperOrigin-RevId: 780935650

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2025-07-09

PiperOrigin-RevId: 780935764

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780936133

allanrenucci committed:
Remove unused `tensorflow` namespace aliases.

We also update the unique usage of `UpdateLogVerbosityIfDefined` to not use the  `tensorflow` qualifier.

PiperOrigin-RevId: 780955815

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 780959099

pifon2a committed:
[XLA:GPU] Use TilingSpace to lookup RTVars for dynamic slice tiling.

PiperOrigin-RevId: 780975832

ZixuanJiang committed:
Add `use_shardy_partitioner` in `tf.tpu.XlaOptions`, `TPUReplicateMetadata`. The default value is false.

PiperOrigin-RevId: 780975971

gflegar committed:
Reduce warning level for incompatible default tile set to info

PiperOrigin-RevId: 780983462

basioli-k committed:
[tfcompile] Improve initialization time for models compiled with the thunk runtime.

Optimizations made
- don't copy constants for initialization
- don't allocate memory for constants (since we're not copying into it)
- don't keep a copy of the function symbols hash map

PiperOrigin-RevId: 780984664

sergey-kozub committed:
PR #28184: Extend `WhileLoopAllReduceCodeMotion` pass with a new pattern (DUS)

Imported from GitHub PR https://github.com/openxla/xla/pull/28184

We're seeing a pattern in Llama3 model (fp8) where all-reduce happens inside the loop, but the results are only used outside of the loop. When there are many participating devices, this could result in a slowdown.

The pass now also matches all-reduces that are scattered into the loop output using "dynamic-update-slice" op with the loop induction variable as the index parameter.
Copybara import of the project:

--
6075de4340b7273c8d63641aeca88fb5ffd7f4a7 by Sergey Kozub <skozub@nvidia.com>:

Extend `WhileLoopAllReduceCodeMotion` pass with a new pattern (DUS)

Merging this change closes #28184

PiperOrigin-RevId: 780999371

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 781000132

tensorflower-gardener committed:
Reverts 24c2972bdcee3d159f4c72d57ff1c7be8a100e54

PiperOrigin-RevId: 781003055

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 781009009

akuegel committed:
[XLA:GPU] Handle nested tuples in FusionCanShareBufferHint.

With DynamicSliceFusion enabled, we can have cases where a fusion has a nested
tuple. So going one step up from the root may not be enough to get to the real
output.
Migrate AliasInfo related tests to their own test target.

PiperOrigin-RevId: 781019728

allanrenucci committed:
Remove `third_party/absl/flags/flag.h` header export in TSL logging library.

And fix transitive dependencies.

PiperOrigin-RevId: 781024653

WillFroom committed:
[XLA:CPU] Don't enable verifier by default.

PiperOrigin-RevId: 781031240

allanrenucci committed:
[XLA:GPU] Use an enum to represent the GPU topology type.

PiperOrigin-RevId: 781045346

Moerafaat committed:
[XLA:GPU/TMA] Extend current autotuner to add TMA optionality to Triton configs.

We also extend block-level and triton-gemm configurations to include a flag for TMA. The flag will be used to enable/disable TMA for all arguments of the kernel, where physically possible.

PiperOrigin-RevId: 781050886

mkuperst committed:
[XLA] Update comment.

`IdenticalIgnoringChannelIdValues` passes `ignore_commutative_operand_order=true`, like `IdenticalIgnoringCommutativeOperandOrder` and unlike `Identical`.

PiperOrigin-RevId: 781053824

mooskagh committed:
[XLA:GPU] [NFC] Move Algebraic Simplifier configs to one place

It was hard to track what simplification runs when, and when you want to enable and disable some transformation, it's hard to find all places you have to touch.

The change is intended to be a NOOP change. The attempt to actually make configs more unified will come in a later change.

PiperOrigin-RevId: 781066663

basioli-k committed:
[XLA][host offllading] Open source host offloading hlo utils

PiperOrigin-RevId: 781072721

basioli-k committed:
[XLA:CPU] Make cpu alignment header public in OSS.

PiperOrigin-RevId: 781077271

WillFroom committed:
[XLA:CPU][XLA:GPU] Add fast min/max rewrite pattern.

PiperOrigin-RevId: 781078198

allanrenucci committed:
Inline unique usage of `tsl::internal::LogString`.

PiperOrigin-RevId: 781088440

tensorflower-gardener committed:
Move xla_workspace4() and xla_workspace3() calls above Python init.

python_init_repositories() loads its own version of bazel_skylib, which causes the http_archive() dep in xla_workspace3() to be ignored.

This is how it's already done in TensorFlow.

PiperOrigin-RevId: 781094000

basioli-k committed:
[tfcompile] Use xla::cpu::Align instead of hardcoded constant.

PiperOrigin-RevId: 781096209

jcai19 committed:
[XLA][Numerics][HLO Value Tracking] Copy original values when creating a clone with same shape

PiperOrigin-RevId: 781108897

derdrdirk committed:
Add AutotunerPass.

The autotuner can be enabled by the --xla_gpu_use_autotuner_pass flag.
The autotuner pass uses the new autotuning backends to autotune hlo instructions.
The autotuner is currently placed next to the GemmFusionAutotuner, but will be moved close to Codegen after validating its functionality and performance.

PiperOrigin-RevId: 781116016

thomasjoerg committed:
[XLA:GPU] Do not fuse sibling transpose fusions with different memory read patterns.

PiperOrigin-RevId: 781120814

dubstack committed:
Don't use position-based matching for entry computations.

PiperOrigin-RevId: 781133140

junwhanahn committed:
Remove the deprecated version of `Array::DisassembleIntoSingleDeviceArrays`

PiperOrigin-RevId: 781138001

pschuh committed:
Add Delete() to CommonPjRtBufferImpl implemented via
AbstractTrackedDeviceBuffer::Delete().

PiperOrigin-RevId: 781147116

tensorflower-gardener committed:
No public description

PiperOrigin-RevId: 781155260

jcai19 committed:
[XLA][Numerics][HLO Value Tracking] Fix printing format of the recovery table

PiperOrigin-RevId: 781162976

WillFroom committed:
[XLA:CPU] Add pass to add loop unroll flags to nested scf.ForOps

PiperOrigin-RevId: 781170081

ecalubaquib committed:
Fix converter copybara removal of learning for oss side

PiperOrigin-RevId: 781176593

pschuh committed:
Expose GetBufferWithHold as a public API.

PiperOrigin-RevId: 781192893

d4n1elchen committed:
[ #HLODiff ] Synchronize scroll for textbox pair

PiperOrigin-RevId: 781203248

GleasonK committed:
[stablehlo] Support bounded dimensions in dot op type inference

PiperOrigin-RevId: 781208712

tensorflower-gardener committed:
Implement LoadedExecutable::GetCompiledMemoryStats() and SizeOfGeneratedCodeInBytes() for proxy.

PiperOrigin-RevId: 781214535

Commit:
[XLA] Add support in collective pipeliner to pipeline forward instructions that are "sink"

Sink means instructions that have some data as input and only output a token, like

data [8,128,128] = data()
cc token[] = custom-call(data)

PiperOrigin-RevId: 781214933

tensorflower-gardener committed:
Solve BUILD deps conflict for linear_solver

PiperOrigin-RevId: 781229194

jcai19 committed:
[XOA][Numerics][HLO Value Tracking] Add a TODO

PiperOrigin-RevId: 781234948

allanrenucci committed:
Fix several LINT errors.

PiperOrigin-RevId: 781235765

ezhulenev committed:
[xla:cpu] Port elemental ir emitter to use XLA intrinsic for f32 to bf16 truncation

PiperOrigin-RevId: 781238590

terryheo committed:
lite: Propagate an error during OpInit()

Delegate kernels may fail during TfLiteRegistration.init(). In this case,
Delegate kernel returns TfLiteKernelInitFailed() to notify. If it happens,
user's call to InterpreterBuilder::operator() or
Interpreter::ModifyGraphWithDelegate() will fail.

Updated SimpleDelegate to use TfLiteKernelInitFailed() instead of nullptr.

Also refactored delegate_test_util to simpler to use.

PiperOrigin-RevId: 781256289

ezhulenev committed:
[xla:codegen] Define XLA Intrinsics: initial commit with documentations and basic functionality

PiperOrigin-RevId: 781272944

pineapplejuice233 committed:
Use the dst device to do the d2d transfer. Instead of waiting for the src buffer definition event, we need to wait for the ready event.

PiperOrigin-RevId: 781275073

vsytch committed:
Allow single core embeddings

PiperOrigin-RevId: 781291822

tensorflower-gardener committed:
[XLA] Add documentation for table lookup pattern.

Adds comments to explain what a "table lookup" is and how it is used in the collective optimization passes.

PiperOrigin-RevId: 781291908

tensorflower-gardener committed:
Expose Auto Sharding to StableHLO.

PiperOrigin-RevId: 781300386

ezhulenev committed:
[xla:codegen] Add Intrinsic::GetOrInsertDeclaration

PiperOrigin-RevId: 781335122

tensorflower-gardener committed:
Do not start XNNPACK fusions with widening converts.

It's better to fuse widening converts into their consumer.

PiperOrigin-RevId: 781346697

nafi3000 committed:
Optimize `xla::HloInstruction::OperandIndices`.

- Don't call `operand_count()` on each iteration. `InlinedVector::size` is more expensive than `vector::size`. It needs a branch first to understand where the storage is.
- Use `InlinedVector::data` and pointer arithmetic instead of `InlinedVector::operator[]` which, again similar to `InlinedVector::size`, needs a branch on each call.
- Motivation for the `s/int64_t/size_t/` change: https://godbolt.org/z/oW59K668a.

PiperOrigin-RevId: 781363727

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781364124

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781367679

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781370948

tensorflower-gardener committed:
Enabling XLA early exit after layout assignment

PiperOrigin-RevId: 781373968

terryheo committed:
Reverts a11dafed1c38656e665f258fb55da77e1c04eefb

PiperOrigin-RevId: 781377322

WillFroom committed:
[XLA:CPU] Add pass to peel loop for workgroups that cross the constraints.

PiperOrigin-RevId: 781427727

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781429676

akuegel committed:
Pass the correct backend-specific AliasInfo to HloMemoryScheduler

The AliasInfo is needed to compute Peak Memory. Theoretically there could be
users of HloMemorySchedule which don't want to compute peak memory, but it
seems not worth it making AliasInfo optional depending on whether peak memory
should be computed.
Adapt direct and indirect users of HloMemoryScheduler accordingly.

PiperOrigin-RevId: 781430019

tensorflower-gardener committed:
Update GraphDef version to 2284.

PiperOrigin-RevId: 781432336

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2025-07-10

PiperOrigin-RevId: 781432355

tensorflower-gardener committed:
LOG(INFO) -> VLOG(4) for layout_canonicalization_callback.

PiperOrigin-RevId: 781436550

WillFroom committed:
[XLA:CPU] Use C style names when required in new fusions.

PiperOrigin-RevId: 781439093

allanrenucci committed:
Manually migrate deprecated references to `strings::StrCat` and `strings::StrAppend`.

`strings::StrCat` and `strings::StrAppend` should eventually forward to `absl::StrCat` and `absl::StrAppend`. Some references need to be rewritten as `absl::StrCat(absl::LegacyPrecision(...))` to avoid loss of precision.

This is a no-op change.

PiperOrigin-RevId: 781446485

beckerhe committed:
Fix unoptimized HLO Snapshots

A recent refactoring has changed the behaviour of the unoptimized HLO Snapshots in such a way that they were dumping optimized HLO. This change rectifies that and adds a test that protects that behaviour.

PiperOrigin-RevId: 781451141

golechwierowicz committed:
[XLA:GPU] Bubble up status handling in unified latency estimator.

PiperOrigin-RevId: 781470886

beckerhe committed:
Introduce ELF section extraction code

This adds code that can extract the contents of a section from an ELF file.

For convenience reason the function also supports static libraries which are
GNU AR archives containing multiple ELF files. In that case the function returns
the first found matching ELF section.

The code can be used to extract .nv_fatbin sections from CUDA libaries and will be
needed to serialize CUDA C++ kernels into serialized GPU programs.

PiperOrigin-RevId: 781482184

WillFroom committed:
[XLA:CPU] Add flag to flatten the call graph after fusion.

PiperOrigin-RevId: 781489163

tensorflower-gardener committed:
Only set `local_device_name_` once in the constructor to avoid data races

PiperOrigin-RevId: 781493257

khasanovaa committed:
[XLA:GPU] Remove prevent mmav3 loop unrolling pass for triton.

This pass was added as a workaround for PTXAS and should have been fixed by NV a long time ago. The benchmarks look good overall and don't introduce regressions on the important models

PiperOrigin-RevId: 781507284

akuegel committed:
Make mutable GetValueSet private (NFC).

Remove the unused variant of this method. Also rename it to GetMutableValueSet
to avoid compile errors, somehow the compiler cannot figure out that it can use
the non-mutable method in certain cases.

PiperOrigin-RevId: 781509391

bixia1 committed:
[xla:layout_assignment] Support buffer types.

Change Shape and ShapeUtil to support the query and modification to the layout of the array storage inside a buffer type.

Enhance layout assignment to support buffer types. There are still problems, such as supporting operand_layout_constraints in layout assignment, which will be addressed in CLs in the sequence.

PiperOrigin-RevId: 781512370

vsytch committed:
[XLA] Make CSE customizable

PiperOrigin-RevId: 781526169

basioli-k committed:
[XLA][host_offloading] Open source host offloading executable.

PiperOrigin-RevId: 781547064

pizzud committed:
hlo_runner_agnostic_test_base: Compile both modules before executing either.

RunAndCompareTwoModules compiles and executes two modules, and if we trigger an
execution before we compile the second then we can't trace the compilations for
caching purposes.

Unfortunately RunAndCompareTwoModulesReplicated doesn't have a nice way to do
the same thing, as decomposing it needs the fundamentally-hardware-dependent
concept of the device assignment.

PiperOrigin-RevId: 781555806

allanrenucci committed:
Fix formatting in `gpu_compiler.cc`.

PiperOrigin-RevId: 781558486

jeffreyadean committed:
Added benchmark for GlobalDecreasingSizeBestFitHeap that exercises
MakeFreeChunks (indirectly by calling Alloc and Free to exercise
different #s of allocated chunks and then calls Finish()).

PiperOrigin-RevId: 781561398

ezhulenev committed:
[xla:codegen] Use FpTrunc::Name to generate implementation

PiperOrigin-RevId: 781563253

jparkerh committed:
Example Extension with cpp dependency injection

Trying to come up with a cleaner way to do dependency injection inside an extension that adds minimal additional apis for those who want to implement directly at the C api. This mechanism keeps the C API cleaner for pure C implementers, but also allows for implementation at CPP for easy composability/testability etc.

PiperOrigin-RevId: 781584176

tensorflower-gardener committed:
Eliminate sdy.data_flow_edge before Shardy->HLO transformation.

PiperOrigin-RevId: 781585969

seantalts committed:
[XLA:CPU] Implement rsqrt with Newton-Raphson refinement.

Design decisions:
- rsqrt is the first and perhaps only function that uses x86-specific intrinsics.
  - We choose the intrinsics we need based on the requested floating point type and vector width.
  - This means we outsource cpu feature detection elsewhere (it will go in the VecDescs generation function).
- Only some vector width + fp types are competitive with 1 / rsqrt.
  - We will benchmark on the main architectures to see where to do the replacement
  - We will add a more holistic hlo fusion benchmark with add, mul, rsqrt to better approximate the cost of the division op.

```
------------------------------------------------------------------------------------------------
Benchmark                                                      Time             CPU   Iterations
------------------------------------------------------------------------------------------------
BM_RsqrtVectorized<4, F32, kRsqrt>/process_time           310412 ns       310495 ns         2271
BM_RsqrtVectorized<4, F32, kOneOverSqrt>/process_time     222650 ns       222991 ns         3125
BM_RsqrtVectorized<8, F32, kRsqrt>/process_time           312053 ns       312342 ns         2243
BM_RsqrtVectorized<8, F32, kOneOverSqrt>/process_time     352200 ns       352706 ns         2010
BM_RsqrtVectorized<8, F64, kRsqrt>/process_time           409729 ns       409382 ns         1714
BM_RsqrtVectorized<8, F64, kOneOverSqrt>/process_time    1178363 ns      1177512 ns          582
```

PiperOrigin-RevId: 781587256

WillFroom committed:
[XLA:CPU] Add conversion from math.erf f64 to libm call

PiperOrigin-RevId: 781590068

bmass02 committed:
Add StatType for input pipeline traces.

PiperOrigin-RevId: 781614433

tensorflower-gardener committed:
Removed an error emission for a failed legalization.

PiperOrigin-RevId: 781616159

mkuperst committed:
[XLA] Factor out the "do channel IDs carry semantic information" check.

Note that `!use_spmd_partitioning_` is likely not the right condition, but I want this to be a pure refactor first and we can adjust the condition later.

PiperOrigin-RevId: 781620792

tensorflower-gardener committed:
Set XNN_FLAG_SLINKY_NO_CHECKS in xnn_fusion_thunk.

PiperOrigin-RevId: 781621183

allanrenucci committed:
[XLA:GPU] Restrict heuristics based collective combiners to multi-host topology.

PiperOrigin-RevId: 781621893

sgarciagoogle committed:
Internal CI changes.

PiperOrigin-RevId: 781622283

ezhulenev committed:
[xla:codegen] Use FpTrunc::CreateDefinition to create xla.fptrunc definitions

PiperOrigin-RevId: 781624007

WillFroom committed:
[XLA:CPU] Add expansion of math.cbrt to pow

PiperOrigin-RevId: 781634299

tensorflower-gardener committed:
Return DeadlineExceededError when getting handler from the pool times out.

PiperOrigin-RevId: 781641050

tensorflower-gardener committed:
Turn off auto sharding stablehlo test.

PiperOrigin-RevId: 781656212

majiddadashi committed:
Fix bounds check for input offsets in quantized ops.

The inputs offsets are negative of the zero points so that should be their acceptable ranges.

PiperOrigin-RevId: 781668766

WillFroom committed:
[XLA:CPU][XLA:GPU] use correct alignment on transfer_write

PiperOrigin-RevId: 781669338

terryheo committed:
Refactor delegate_test_util to simpler to use

This change improves the usability of the test_utils::SimpleDelegate utility
by replacing the long list of boolean parameters with readable Options enum.

PiperOrigin-RevId: 781681590

bixia1 committed:
[xla:gpu] Handle buffer related custom calls.

Add executable tests.

PiperOrigin-RevId: 781699123

bixia1 committed:
[XLA] Implement operand-to-result layout propagation for custom calls

This allows propagating layout from an operand to the result, based on output-to-operand-alias information.

PiperOrigin-RevId: 781706746

mkuperst committed:
[XLA] CSE collectives with different channel IDs when it's safe to do so.

PiperOrigin-RevId: 781721931

nvgrw committed:
Reduce likelihood of collisions in precompilation using Default print options.

PiperOrigin-RevId: 781724957

ddunl committed:
Migrate uses of `XLA_TEST_BACKEND` macros to use utilities in `xla_test_backend_predicates.h`

PiperOrigin-RevId: 781737406

ddunl committed:
Move `tensorflow/third_party/llvm` to `xla/third_party/llvm`

PiperOrigin-RevId: 781763691

juliagmt-google committed:
[XLA:benchmarks] Remove comments related to testing

PiperOrigin-RevId: 781766386

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781814264

tensorflower-gardener committed:
Reverts 1d1309b2712b001b325962ee76972f14681913da

PiperOrigin-RevId: 781828585

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781832145

ezhulenev committed:
[xla:cpu] Remove XLA_ALIGN macro

PiperOrigin-RevId: 781843003

tensorflower-gardener committed:
Make `Execute*()` `const` for `*Executable` classes.

We cache `*Executable` objects to reuse compilation results. This means that we may call `Execute*()` on such cached objects multiple times and expect that the calls won't affect the objects' behaviors. In other words, `*Executable()` should be `const`. Marking them `const` prevents the code from accidentally mutating cached executables and causing bugs.

PiperOrigin-RevId: 781847420

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781858106

jerryxyj committed:
No public description

PiperOrigin-RevId: 781861930

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 781875010

WillFroom committed:
[XLA:CPU] Use xla intrinsic f32 -> bf16 conversion.

PiperOrigin-RevId: 781883075

tensorflower-gardener committed:
Update GraphDef version to 2285.

PiperOrigin-RevId: 781890029

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2025-07-11

PiperOrigin-RevId: 781890727

WillFroom committed:
[XLA:CPU] Fix race condition in LowerXlaMathLibPass.

PiperOrigin-RevId: 781894153

allanrenucci committed:
[XLA:GPU] Enable heuristic based synchronous and pipelined collective combining by default.

This change only takes effect if the user does not set the `xla_gpu_{all_reduce,all_gather,reduce_scatter}_combine_threshold_bytes` flags.

PiperOrigin-RevId: 781902147

KanishAnand committed:
Use unused sharding variable

PiperOrigin-RevId: 781903248

akuegel committed:
Remove mutable buffer accessor methods from HloAliasAnalysis (NFC).

All current users don't actually need mutable access.

PiperOrigin-RevId: 781905323

WillFroom committed:
[XLA:CPU] Add log1p math lib function.

PiperOrigin-RevId: 781926524

akuegel committed:
Remove dead code from TuplePointsToAnalysis (NFC).

While there, also fix a few ClangTidy warnings.

PiperOrigin-RevId: 781947527

basioli-k committed:
Reverts 55a3ba02a6b51595dea5efaf3dce7e44ab2409c1

PiperOrigin-RevId: 781950054

tensorflower-gardener committed:
Move llvm() and llvm_setup() calls down one workspace*.bzl file.

This makes XLA consistent with TF.

PiperOrigin-RevId: 781955402

chsigg committed:
[Triton] Add triton squeeze dims pass

This pass removes size-1 dimensions from tensors to generate faster code
through Triton.

- `squeeze_dims` ops are extracted from each unit dimension of `tt.reshape`.
- A series of patterns push `squeeze_dims` up through the graph, past element-wise ops, broadcast, transpose, join, reduce.
- `squeeze_dims` are folded into `tt.load` and `tt.expand_dims`, or converted back to `tt.reshape`.

PiperOrigin-RevId: 781962093

amd-songpiao committed:
PR #28314: [ROCm] added allreduce kernel registration

Imported from GitHub PR https://github.com/openxla/xla/pull/28314

added the missing allreduce kernel registration on ROCm

it fixes all the failed AllReduceTest/AllReduceTest.* in collective_ops_e2e_test. e.g.
`bazel-bin/xla/tests/collective_ops_e2e_test_amdgpu_any --gtest_filter=AllReduceTest/AllReduceTest.AsyncAllReduce_F32_2GPUs/async_one_shot`

@xla-rotation could you review my PR, please?
Copybara import of the project:

--
6517bd88291374e9754838ac82a5d58fe4386c8d by songlin <Songlin.Piao@amd.com>:

adapt allreduce kernel registration in rocm

--
6e255aa68932a42964c048dd24ecd944a7dbf0c9 by songlin <Songlin.Piao@amd.com>:

adapt to get rid of macro usage

--
af96c9743ad6abbe323709d1721c013abb84e206 by songlin <Songlin.Piao@amd.com>:

make PutSignalFlag and WaitSignalFlag as templates with specialization for ROCm and CUDA

Merging this change closes #28314

PiperOrigin-RevId: 781992142

tensorflower-gardener committed:
Reverts 9b88203f15334e7e779bb2b3200448a07ad0245b

PiperOrigin-RevId: 781999343

metaflow committed:
[XLA:GPU] flag to control nested gemm fusion

For more granular control over generic triton emitter.

That replaces the binary `xla_gpu_unsupported_enable_generic_triton_emitter_for_gemms`.

Sample usage `--xla_gpu_unsupported_generic_triton_emitter_features=enable_nested_gemm`

Drive by update of the comment on xla_gpu_enable_llvm_module_compilation_parallelism.

PiperOrigin-RevId: 782014919

apivovarov committed:
[Autotuner] Add block level emitter backend for Triton fusion.

This change introduces the Triton block-level fusion emitter backend, which enables autotuning of tile configurations for custom Triton fusions in XLA.

This backend implements the following core interfaces:
- GetSupportedConfigs:
Enumerates all supported combinations of tile sizes for the output tensors. The generated configs can be used during autotuning to explore different performance candidates. (will be added in the next PR)
- GetDefaultConfig:
Provides a default tile configuration for a given Triton fusion, used as a fallback when no tuning data is available.
- ApplyConfig:
Applies a selected block-level fusion configuration to a Triton fusion instruction by updating its GpuBackendConfig. (will be added in the next PR)

PiperOrigin-RevId: 782032163

zacmustin committed:
Fix clang-tidy errors in `pjrt_c_api_helpers.cc`.

PiperOrigin-RevId: 782047204

seantalts committed:
[XLA:CPU] Refactor Rsqrt intrinsic with new UnaryIntrinsicBase.

PiperOrigin-RevId: 782077991

tensorflower-gardener committed:
Add traceme for IFRT executable launch

PiperOrigin-RevId: 782078942

ezhulenev committed:
[xla:cpu] Move memory allocation functions to tf2xla

PiperOrigin-RevId: 782080412

tensorflower-gardener committed:
[XLA] Refactor HloDCE for improved readability.

A minor modification is also made for dangling computations where `changed` is only set to `true` when a computation is actually removed to account for the possibility that a dangling computation might be filtered by execution_threads.

PiperOrigin-RevId: 782086875

ezhulenev committed:
[xla:codegen] Start migrating Ldexp to Intrinsic declaration

PiperOrigin-RevId: 782093202

GleasonK committed:
[SDY] Refactor MHLO dependencies where possible

PiperOrigin-RevId: 782093218

ddunl committed:
Migrate uses of `XLA_TEST_BACKEND` macros to use utilities in `xla_test_backend_predicates.h`

Reverts c9ca1c456e1032ba1e4d728fdda68bf9a46b4f92

PiperOrigin-RevId: 782094674

ddunl committed:
Now that we don't steal from `tensorflow/third_party` don't do any post hoc Copybara for TensorFlow builds

(*) Except for `tensorflow/third_party/py`

PiperOrigin-RevId: 782106216

tensorflower-gardener committed:
Reverts e4fccd6471306e8b4d4012a423376c170e6a6301

PiperOrigin-RevId: 782126042

SiqiaoWu1993 committed:
Fix sink invariant in tf.ifregion.

PiperOrigin-RevId: 782135828

sdasgup3 committed:
[Phase Compilation] Part-4: Add C++ layers to test and interact with C PJRT API.

PiperOrigin-RevId: 782147317

seantalts committed:
[XLA:CPU] Use templated intrinsic helpers for log1p and fix intrinsic name.

PiperOrigin-RevId: 782156735

nafi3000 committed:
Optimize `GlobalDecreasingSizeBestFitHeap::MakeFreeChunks` by 1.2X to 1.4X.

`BM_GlobalDecreasingSizeBestFitHeap` benchmark is optimized by up to 1.17X.

PiperOrigin-RevId: 782157151

xinxinmo committed:
[xla] Add back test targets in //xla/tools/hlo_diff on CI because the service outage

errors (https://github.com/openxla/xla/actions/runs/16007180610/job/45156461568)

have been fixed in https://github.com/openxla/xla/pull/28465.

Reverts f590575b0ed93e703ec30b29f815bf5077ce5468

PiperOrigin-RevId: 782158057

ICGog committed:
[IFRT IR] Use the auto-generate pass creation functions rather than custom defined funcs.

PiperOrigin-RevId: 782172174

felixwqp committed:
[XLA]  Extract `ExtractSplitDimSpec` from `MatchWithDynamicSlice` in collective_opt_utils.

PiperOrigin-RevId: 782183029

ezhulenev committed:
[xla:codegen] Distinguish between scalar and vector arguments/results in XLA intrinsics

PiperOrigin-RevId: 782184041

pschuh committed:
Add CommonPjRtRawBuffer::ScheduleCopyTo which is an async version of
CommonPjRtRawBuffer::CopyTo. This allows scheduling the copy async after
some dependencies have completed. Backends which have to allocate resources
to perform copies may want to override this instead to garuntee ordering.

PiperOrigin-RevId: 782185871

emilyfertig committed:
[JAX] Add an interface for cross-host DCN transfers and a PjRt-IFRT/Linux implementation

PiperOrigin-RevId: 782197772

ezhulenev committed:
[xla:codegen] Introduce Intrinsic::Type class instead of an alias

PiperOrigin-RevId: 782199950

ICGog committed:
[IFRT IR] Add pass for dumping IFRT IR graphs to dot representation

PiperOrigin-RevId: 782216238

slackito committed:
Integrate LLVM at llvm/llvm-project@f8cb7987c64d

Updates LLVM usage to match
[f8cb7987c64d](https://github.com/llvm/llvm-project/commit/f8cb7987c64d)

PiperOrigin-RevId: 782220044

tensorflower-gardener committed:
Automated Code Change

PiperOrigin-RevId: 782239761

tensorflower-gardener committed:
Update GraphDef version to 2286.

PiperOrigin-RevId: 782276736

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2025-07-12

PiperOrigin-RevId: 782276814

tensorflower-gardener committed:
Fix typo in xnn_fusion_thunk.cc.
Should use ifdef rather than ifndef.

PiperOrigin-RevId: 782343552

tomnatan30 committed:
#sdy If auto partitioning is enabled and there is no registered auto partitioner, register Alpa as the default.

PiperOrigin-RevId: 782366951

tensorflower-gardener committed:
Update GraphDef version to 2287.

PiperOrigin-RevId: 782530395

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2025-07-13

PiperOrigin-RevId: 782530445

chsigg committed:
Reverts 6503034148ab3c0469a32d20b9a3ea397457a8f8

PiperOrigin-RevId: 782532723

ezhulenev committed:
[xla:codegen] Use Intrinsic::Type in Fptruc::GetOrInsertDeclaration

PiperOrigin-RevId: 782601860

pineapplejuice233 committed:
Always stage transfers when doing d2h copy to avoid memory corruption issue.

PiperOrigin-RevId: 782740753

amd-jianli12 committed:
PR #28401: [ROCm] Fix PackedTranspose for adapting to warp size 64

Imported from GitHub PR https://github.com/openxla/xla/pull/28401

Calculate ranges for dimensions in IndexingMap with kTileSize rather than WarpSize() since tile size is still expected to be vector_size * kNumShmemBanks which we preserve as 32.

Fixed failing test:
[  FAILED  ] Convolve2D_1x2x2x1024_2x2x128x512_Grouped_Valid/0.Types, where TypeParam = Eigen::half
Copybara import of the project:

--
33e7ad3c96584471ff9b8732f6fdecf04ef9ed84 by Jian Li <Jian.Li@amd.com>:

[ROCm] Fix PackedTranspose for adapting to warp size 64

Calculate ranges for dimensions in IndexingMap with kNumShmemBanks
rather than WarpSize() since tile size is still expected to be
vector_size * kNumShmemBanks which we preserve as 32.

Fixed failing test:
[  FAILED  ] Convolve2D_1x2x2x1024_2x2x128x512_Grouped_Valid/0.Types, where TypeParam = Eigen::half

Merging this change closes #28401

PiperOrigin-RevId: 783204421

mmakevic-amd committed:
Merge branch 'upstream/master' into develop-upstream-sync-250714

mmakevic-amd committed:
Fix merge conflicts

mmakevic-amd committed:
Remove everything from /third_party/triton and unnecessary BUILD files from xla/third_party/triton

mmakevic-amd committed:
Use cuda_library instead of obsolete gpu_kernel_library

mmakevic-amd committed:
Remove unnecessary include

mmakevic-amd committed:
Add multi gpu skip tags to cpu.bazelrc

mmakevic-amd committed:
Fix run_cpu.sh to be in accordance with Bazel test command used in CI

mmakevic-amd committed:
Skip test using compute_compability instead of macro

mmakevic-amd committed:
Disable failing XLA tests

mmakevic-amd committed:
Re-disable //xla/service/gpu/tests:single_instruction.hlo.test

mmakevic-amd committed:
Fix gpu_kernel_tiling_test

mmakevic-amd committed:
Disable MultiOutputFusionTest.MultiOutputReduceFusionMajorWithExtraOutput subtest

mmakevic-amd committed:
Disable gpu_hlo_llvm.hlo

mmakevic-amd committed:
Disable failing subtests in triton_fusion_numerics_verifier_test

mmakevic-amd committed:
Disable failing hlo tests

mmakevic-amd committed:
Enable one of the gpu_compiler_tests

mmakevic-amd committed:
Disable test using compute capability instead of obsolete macro

mmakevic-amd committed:
Enable TanTest. Use TEST_F instead of XLA_TEST_F

hsharsha reviewed (approved):
Checked merge conflict resolution and commits after that. LGTM

mmakevic-amd commented:
> @mmakevic-amd could you have a list of disabled and fixed UTs on this weekly-sync PR description?

Fixed. All of these tests are failing in the upstream as well
